src/routers/item.js

```diff
@@ -0,0 +1,23 @@
+const express = require('express');
+
+const router = express.Router();
+
+const itemController = require('../controllers/item');
+
+// GET route for /available
+router.get('/available', itemController.getAvailableItems);
+
+// GET, PUT and DELETE routes for /:id
+router
+  .route('/:id')
+  .get(itemController.getSingleItem)
+  .put(itemController.updateItem)
+  .delete(itemController.deleteItem);
+
+// GET route for /filter
+router.get('/filter', itemController.getFilteredItems);
```

Comment:
I think you should move the `/filter` route above `/:id` otherwise there can be issues in the control flow of the routes.

Sometimes what happens is that the Express router will look at the routes one by one, and if `/:id` comes first it can think that "filter" is an ID value matching this route and then the control will never go to your next controller. You can even test this out on postman to double-check, that which function gets executed if `/:id` comes before `/filter` and the `/filter` API is called.

Reply:
Yes, makes sense. I remembered we had faced that before. Thank you.